homeworlds 1.1.0

samford opened the pull request:
<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

~The existing `stable` URL for `homeworlds` is a Git revision with no `tag`. We have some other formulae in a similar situation but they use a commit tarball instead of checking out the Git repository (e.g., `luajit`), so this updates the formula to follow the same pattern.~

I asked upstream if they would be willing to tag versions and they created v1.0.0 and v1.1.0 tags in response. This updates the formula to 1.1.0, switching the formula to a tag tarball in the process.

p-linnane commented:
What other formula are doing this? The warning in the `luajit` formula sounds like we should not do this. https://github.com/Homebrew/homebrew-core/blob/dde579ee4f5652dbeea8c6c24cd41899dd26f755/Formula/l/luajit.rb#L1-L5

samford commented:
> What other formula are doing this?

It's four old formulae that have been using a specific commit with no tag since they were introduced:

* colortail: Originally added in https://github.com/Homebrew/legacy-homebrew/pull/40428; 0.3.4 was committed on 2012-08-01 but never tagged (see https://github.com/Homebrew/homebrew-core/pull/205260#issuecomment-2610866700)
* fabric-completion: Added to homebrew/core in https://github.com/Homebrew/homebrew-core/pull/9198/files#diff-2dfe78dbb8badfe2a0572236edf4bc9231d90259cd14eb4b40e8d582eb0b555d; unversioned, no tags or releases
* homeworlds: Added to homebrew/core in https://github.com/Homebrew/homebrew-core/pull/9685/files#diff-7949d1dfdf9cbbd742e36fc3ddb2ad8e92397fd30a7ce8a1fe324dc0667bcfa5; unversioned, no tags or releases
* pip-completion: Added to homebrew/core in https://github.com/Homebrew/homebrew-core/pull/9198/files#diff-6348859406796f4b0ea930d70fdaba24ad5f174f747d4770e930f453ce956c40; unversioned, no tags or releases

The comment in `luajit` is correct but these were added before that point and we can't use a tag in these cases. I've been updating these to use a commit tarball, as we only use Git when there's a technical need and that doesn't appear to be the case for these (they're just holdovers from years past). These don't align with our current standards but using a tarball is a little better than the status quo.

ZhongRuoyu reviewed (commented):
The change looks OK to me per se since we are already using the commit so it doesn't worsen the situation. However I do think that we should deprecate this formula since it doesn't really meet [our standards](https://docs.brew.sh/Acceptable-Formulae#stable-versions) ([and is not popular either](https://formulae.brew.sh/formula/homeworlds)). (Same for the other formulae mentioned in https://github.com/Homebrew/homebrew-core/pull/205524#issuecomment-2614186553 unless there's a reason not to.)

samford commented:
Works for me. I've updated this to deprecate the formula.

I'm planning to open an upstream issue for `colortail`, to see if they will tag 0.3.4. A former Homebrew maintainer asked them to tag releases (ten years ago) and they tagged 0.3.3 but didn't go on to tag 0.3.4. It seems like they may be receptive if we ask again and that would allow us to switch the formula to a tag tarball instead of deprecating it.

I can handle the deprecation for `pip-completion` in the upcoming PR to update the `stable URL. I'll create PRs for the others (outside of `colortail` and `wasi-libc`) afterward.

carlocab commented on Formula/h/homeworlds.rb:
```diff
@@ -25,6 +25,9 @@ class Homeworlds < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "88f7a3976037aea175f742cb16c1e3e0e7e0f7945bf9f24b42eaedeabe16834c"
   end
 
+  # Unversioned and upstream repository has no tags or releases.
+  deprecate! date: "2025-01-26", because: "does not meet requirements for `stable` URLs"
```
Can we ask upstream if they're willing to cut a release tag before deprecating this?

samford replied:
Sure, let's see if we get a response: https://github.com/Quuxplusone/Homeworlds/issues/5

That said, it's worth acknowledging that we're asking upstream to version an unversioned project and that may not be something that they want to do (especially considering it hasn't been updated in a few years). There's no harm in asking but we should temper our expectations for these projects.

samford commented:
In response to my issue, upstream tagged the commit we've been using as v1.0.0 and also created a v1.1.0 tag for subsequent changes. I've updated the formula to use the v1.1.0 tag tarball. It's necessary to bump `version_scheme` since we were using a date version, so 1.1.0 will be seen as lower than 20141022.

Upstream renamed the binaries in 1.1.0, so we can simplify the `install` steps a little. I also added a simple test for the CLI (there don't appear to be any non-interactive commands). Let me know if there's any room for improvement here.

carlocab reviewed (approved):
Thanks!
